feat: Update Loki monitoring docs to new meta monitoring helm (#13176)
grafana/loki: docs/sources/setup/install/helm/monitor-and-alert/with-grafana-cloud.md

These instructions are based off the [meta-monitoring-chart repository](https://github.com/grafana/meta-monitoring-chart/tree/main).

## Before you begin

- Helm 3 or above. See [Installing Helm](https://helm.sh/docs/intro/install/).
- A Grafana Cloud account and stack (including Cloud Grafana, Cloud Metrics, and Cloud Logs).
- A running Loki deployment installed in that Kubernetes cluster via the Helm chart.

## Configure the meta namespace

The meta-monitoring stack will be installed in a separate namespace called `meta`. To create this namespace, run the following command:

  ```bash
    kubectl create namespace meta
  ```
  
## Grafana Cloud Connection Credentials

The meta-monitoring stack sends metrics, logs, and traces to Grafana Cloud. This requires that you know your connection credentials to Grafana Cloud. To obtain connection credentials, follow the steps below:

1. Create a new Cloud Access Policy in Grafana Cloud.     
    1. Sign into [Grafana Cloud](https://grafana.com/auth/sign-in/).

>     1.  In the main menu, select **Administration > Users and Access > Cloud Access Policies**.
>     1. Click **Create access policy**.
>     1. Give the policy a **Name** and select the following permissions:
>       - Logs: Write
>       - Metrics: Write
>       - Traces: Write
>   1. Click **Create**.

Reviewer: ```suggestion
    1. In the main menu, select **Security > Access Policies**.
    1. Click **Create access policy**.
    1. Give the policy a **Name** and select the following permissions:
       - Metrics: Write
       - Logs: Write
       - Traces: Write
  1. Click **Create**.
```

  - grafana cloud
---

# Monitor Loki with Grafana Cloud

> This guide will walk you through using Grafana Cloud to monitor a Loki installation set up with the Helm chart. This method takes advantage of many of the chart's self-monitoring features, sending metrics, logs, and traces from the Loki deployment to Grafana Cloud. Monitoring Loki with Grafana Cloud offers the added benefit of troubleshooting Loki issues even when the Helm-installed Loki is down, as the telemetry data will remain available in the Grafana Cloud instance.

Reviewer: ```suggestion
This guide will walk you through using Grafana Cloud to monitor a Loki installation set up with the `meta-monitoring` Helm chart. This method takes advantage of many of the chart's self-monitoring features, sending metrics, logs, and traces from the Loki deployment to Grafana Cloud. Monitoring Loki with Grafana Cloud offers the added benefit of troubleshooting Loki issues even when the Helm-installed Loki is down, as the telemetry data will remain available in the Grafana Cloud instance.
```

1. Click the **Details** button for your Loki instance.
     1. From the **Using Grafana with Logs** section, collect the instance **Name** and **URL**.
     1. Navigate back to the **Overview** page.
1. Click the **Details** button for your Tempo instance.
    1. From the **Using Grafana with Tempo** section, collect the instance **Name** and **URL**.

3. Finally, generate the secrets to store your credentials for each metric type within your Kubernetes cluster:
   ```bash
      kubectl create secret generic logs -n meta \
        --from-literal=username=<USERNAME LOGS> \
        --from-literal= <ACCESS POLICY TOKEN> \
        --from-literal=endpoint='https://<LOG URL>/loki/api/v1/push'

        kubectl create secret generic metrics -n meta \
        --from-literal=username=<USERNAME METRICS> \
        --from-literal=password=<ACCESS POLICY TOKEN> \
        --from-literal=endpoint='https://<METRICS URL>/api/prom/push'

        kubectl create secret generic traces -n meta \
        --from-literal=username=<OTLP INSTANCE ID> \
        --from-literal=password=<ACCESS POLICY TOKEN> \
        --from-literal=endpoint='https://<OTLP URL>/otlp'
   ```

## Configuration and Installation

> To install the meta-monitoring helm chart, you must create a `values.yaml` file. At a minimum this file should contain the following:

Reviewer: ```suggestion
To install the `meta-monitoring` Helm chart, you must create a `values.yaml` file. At a minimum this file should contain the following:
```

  * The namespace to monitor
  * Enablement of cloud monitoring

This example `values.yaml` file provides the minimum configuration to monitor the `loki` namespace:

```yaml
  namespacesToMonitor:
  - default

  cloud:
    logs:
      enabled: true
      secret: "logs"
    metrics:
      enabled: true
      secret: "metrics"
    traces:
      enabled: true
      secret: "traces"
```
For further configuration options, refer to the [sample values.yaml file](https://github.com/grafana/meta-monitoring-chart/blob/main/charts/meta-monitoring/values.yaml).

> To install the meta-monitoring helm chart, run the following commands:

Reviewer: ```suggestion
To install the `meta-monitoring` Helm chart, run the following commands:
```

        - record: cluster_namespace_job_route:loki_request_duration_seconds:50quantile
          expr: histogram_quantile(0.50, sum(rate(loki_request_duration_seconds_bucket[5m])) by (le, cluster, namespace, job, route))
        - record: cluster_namespace_job_route:loki_request_duration_seconds:avg
          expr: sum(rate(loki_request_duration_seconds_sum[5m])) by (cluster, namespace, job, route) / sum(rate(loki_request_duration_seconds_count[5m])) by (cluster, namespace, job, route)
        - record: cluster_namespace_job_route:loki_request_duration_seconds_bucket:sum_rate
          expr: sum(rate(loki_request_duration_seconds_bucket[5m])) by (le, cluster, namespace, job, route)
        - record: cluster_namespace_job_route:loki_request_duration_seconds_sum:sum_rate
          expr: sum(rate(loki_request_duration_seconds_sum[5m])) by (cluster, namespace, job, route)
        - record: cluster_namespace_job_route:loki_request_duration_seconds_count:sum_rate
          expr: sum(rate(loki_request_duration_seconds_count[5m])) by (cluster, namespace, job, route)
   ```

> ## Kube-state-metrics

Reviewer: ```suggestion
## Install kube-state-metrics
```
